Source/WTF/wtf/FileSystem.cpp

```diff
@@ -365,7 +363,7 @@ bool MappedFileData::mapFileHandle(PlatformFileHandle handle, FileOpenMode openM
     if (data == MAP_FAILED)
         return false;
 
-    m_fileData = { static_cast<uint8_t*>(data), size };
+    m_fileData = unsafeMakeSpan(static_cast<uint8_t*>(data), size);
```

Comment:
It would be even better to use `MallocSpan` here to guarantee `munmap` gets called.

Reply:
There are issues with reusing `MallocSpan` for this:
- Malloc::free() currently only take a pointer, no size (Malloc being the template parameter)
- Malloc::malloc() only takes a size but `mmap()` would take a lot more parameters.

I agree with should have some sort of RAII object to avoid memory leaks. However, I'm not sure reusing MallocSpan is suitable here. What do you think?

Reply:
I don’t really care much. `MappedFileData` itself is a sort of RAII object, just has too much other code. So maybe we just need to refactor into two classes to separate concerns better. Seems like in theory a template like `MallocSpan` could be designed to be flexible about the arguments to allocation and include the size for the free function, since the size is what makes a span. But I understand that’s not how the actual template is designed, and that’s OK.